src/core/Logger.cpp

```diff
@@ -4,6 +4,15 @@
 
 #include "Logger.h"
 
+#ifdef _WIN32
+#ifndef WIN32_LEAN_AND_MEAN
+#  define WIN32_LEAN_AND_MEAN
```

Comment:
I'd still rather this LEAN_AND_MEAN stuff be done in CMake itself, rather than adding it to a random source file. [See how we did it in KVIrc](https://github.com/kvirc/KVIrc/blob/master/CMakeLists.txt#L973)